taxi/tests/test_forms.py

```diff
@@ -0,0 +1,141 @@
+        self.user = get_user_model().objects.create_user(
+            username="TestUsername",
+            password="TestPassword123"
+        )
+        self.client.force_login(self.user)
+        self.manufacturer1 = Manufacturer.objects.create(name="Toyota", country="Japan")
+        self.manufacturer2 = Manufacturer.objects.create(name="Ford", country="USA")
+        self.manufacturer3 = Manufacturer.objects.create(name="Tesla", country="USA")
+
+    def test_manufacturer_search_by_name(self):
+        response = self.client.get(reverse("taxi:manufacturer-list") + '?name=Toyota')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Toyota")
+        self.assertNotContains(response, "Ford")
+        self.assertNotContains(response, "Tesla")
+
+    def test_manufacturer_search_by_partial_name(self):
+        response = self.client.get(reverse("taxi:manufacturer-list") + '?name=Tes')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Tesla")
+        self.assertNotContains(response, "Toyota")
+        self.assertNotContains(response, "Ford")
+
+    def test_manufacturer_search_no_results(self):
+        response = self.client.get(reverse("taxi:manufacturer-list") + '?name=BMW')
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, "Toyota")
+        self.assertNotContains(response, "Ford")
+        self.assertNotContains(response, "Tesla")
+
+
+class DriverSearchTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="TestUsername",
+            password="TestPassword123"
+        )
+        self.client.force_login(self.user)
+        self.driver1 = get_user_model().objects.create_user(
+            username="Driver1",
+            password="TestPassword123",
+            first_name="Test",
+            last_name="Driver1",
+            license_number="AAA11111"
+        )
+        self.driver2 = get_user_model().objects.create_user(
+            username="Driver2",
+            password="TestPassword123",
+            first_name="Test",
+            last_name="Driver2",
+            license_number="AAA22222"
+        )
+        self.driver3 = get_user_model().objects.create_user(
+            username="Driver3",
+            password="TestPassword123",
+            first_name="Test",
+            last_name="Driver3",
+            license_number="AAA33333"
+        )
+
+    def test_driver_search_by_username(self):
+        response = self.client.get(reverse("taxi:driver-list") + '?username=Driver1')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Driver1")
+        self.assertNotContains(response, "Driver2")
+        self.assertNotContains(response, "Driver3")
+
+    def test_driver_search_by_partial_username(self):
+        response = self.client.get(reverse("taxi:driver-list") + '?username=Driver2')
```

Comment:
The test name `test_driver_search_by_partial_username` suggests that it should test partial matches, but the query `?username=Driver2` is a full username. Consider changing the query to a partial username to match the test name.